sale_timesheet_rounded/models/account_analytic_line.py

```diff
@@ -0,0 +1,166 @@
+        # we must check if the value is there or not.
+        # If you set a default, you are going to get it as value at create
+        # and the computation won't happen.
+        # default=0.0,
+        copy=False,
+    )
+
+    @api.model
+    def _init_column(self, column_name):
+        """ Initialize the value of the given column for existing rows.
+            Overridden here because we need to have different default values
+            for unit_amount_rounded for every analytic line.
+        """
+        if column_name != "unit_amount_rounded":
+            super()._init_column(column_name)
+        else:
+            _logger.info(
+                "Initializing column `unit_amount_rounded` with the "
+                "value of `unit_amount`"
+            )
+            query = 'UPDATE "%s" SET unit_amount_rounded = unit_amount;' % self._table
+            self.env.cr.execute(query)
+
+    @api.onchange("unit_amount")
+    def _onchange_unit_amount(self):
+        self.unit_amount_rounded = self.with_context(
+            force_compute=1
+        )._calc_unit_amount_rounded()
+
+    def _calc_unit_amount_rounded(self):
+        self.ensure_one()
+        force_compute = self.env.context.get("force_compute")
+        # TODO: do we still need this?
+        self = self.with_context(timesheet_rounding=True)
+        project_rounding = (
+            self.project_id and self.project_id.timesheet_rounding_method != "NO"
+        )
+        already_set = self.unit_amount_rounded and not force_compute
+
+        if project_rounding and not already_set:
+            return self._calc_rounded_amount(
+                self.project_id.timesheet_rounding_unit,
+                self.project_id.timesheet_rounding_method,
+                self.project_id.timesheet_rounding_factor,
+                self.unit_amount,
+            )
+        else:
+            return self.unit_amount
+
+    @staticmethod
+    def _calc_rounded_amount(rounding_unit, rounding_method, factor, amount):
+        factor = factor / 100.0
+        if rounding_unit:
+            unit_amount_rounded = float_round(
+                amount * factor,
+                precision_rounding=rounding_unit,
+                rounding_method=rounding_method,
+            )
+        else:
+            unit_amount_rounded = amount * factor
+        return unit_amount_rounded
+
+    def _timesheet_postprocess(self, values):
+        # Add no rounding key to avoid triggering
+        # _update_unit_amount_rounded on the write from
+        # _timesheet_postprocess to update the amount
+        self = self.with_context(_no_rounding=True)
+        return super()._timesheet_postprocess(values)
+
+    ####################################################
+    # ORM Overrides
+    ####################################################
+
+    @api.model
+    def create(self, values):
+        res = super().create(values)
+        # TODO improve me
```

Comment:
Can you check this TODO ?

Reply:
same answer https://github.com/OCA/timesheet/pull/352#discussion_r429764937

Reply:
thanks